src/v/kafka/server/server.cc

```diff
@@ -265,10 +265,16 @@ ss::future<> server::apply(ss::lw_shared_ptr<net::connection> conn) {
         config::shard_local_cfg().enable_sasl());
     const auto authn_method = get_authn_method(*conn);
 
+    const auto conn_max_reauth_ms
+      = config::shard_local_cfg().connections_max_reauth_ms();
+
+    vlog(klog.debug, "max_reauth_ms: {}", conn_max_reauth_ms);
+
     // Only initialise sasl state if sasl is enabled
     auto sasl = authn_method == config::broker_authn_method::sasl
                   ? std::make_optional<security::sasl_server>(
-                    security::sasl_server::sasl_state::initial)
+                    security::sasl_server::sasl_state::initial,
+                    conn_max_reauth_ms)
```

Comment:
an argument for making this max parameter be a `config::binding<ms>` is that since the configuration is locked in when the connection is created, changes to the max reauth only take affect for new connections. it's probably not a big deal, but in principle someone might want to lower the max reauth but wouldn't have a way to forcefully shutdown a client connection without restarting redpanda.

Reply:
> Not sure about this one.

I think it is probably ok as-is. What I'd do is just give a little thought to the possible scenarios that might occur and if any of those scenarios have the property that we wish it were live updatable in the context of an incident.